homeassistant/components/reolink/views.py

```diff
@@ -0,0 +1,138 @@
+"""Reolink Integration views."""
+
+from __future__ import annotations
+
+from http import HTTPStatus
+import logging
+
+from aiohttp import ClientError, ClientTimeout, web
+from reolink_aio.enums import VodRequestType
+from reolink_aio.exceptions import ReolinkError
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util.ssl import SSLCipherList
+
+from .util import get_host
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def async_generate_playback_proxy_url(
+    config_entry_id: str, channel: int, filename: str, stream_res: str, vod_type: str
+) -> str:
+    """Generate proxy URL for event video."""
+
+    url_format = PlaybackProxyView.url
+    return url_format.format(
+        config_entry_id=config_entry_id,
+        channel=channel,
+        filename=filename.replace("/", "|"),
```

Comment:
I don't like custom escaping, as we can easily forget a character.
We should use URL encoding or maybe even better encoding all required data in base64

Reply:
Good suggestion, I am now using the urllib.parse.quote and urllib.parse.unquote functions.